src/encode.rs

```diff
@@ -68,6 +68,41 @@ impl EncodeTarget for Vec<u8> {
     }
 }
 
+#[cfg(feature = "smallvec")]
+impl<A: smallvec::Array<Item = u8>> EncodeTarget for smallvec::SmallVec<A> {
+    /// Encodes data into a [`smallvec::SmallVec`].
+    ///
+    /// Note that even if the encoded value fits into vector’s inline buffer,
+    /// this may result in allocation if `max_len` is greater than vector’s
+    /// inline size.  To make sure that the inline buffer is enough for N-byte
+    /// buffer encoded in base58, use smallvec with ⌈N*1.5⌉-byte long inline
+    /// buffer (or ⌈(N+5)*1.5⌉ if version and checksum are included).
+    fn encode_with(
+        &mut self,
+        max_len: usize,
+        f: impl for<'a> FnOnce(&'a mut [u8]) -> Result<usize>,
+    ) -> Result<usize> {
+        struct Guard<'a, A: smallvec::Array> {
+            vec: &'a mut smallvec::SmallVec<A>,
+            len: usize,
+        }
+
+        impl<'a, A: smallvec::Array> std::ops::Drop for Guard<'a, A> {
+            fn drop(&mut self) {
+                unsafe { self.vec.set_len(self.len) }
+            }
+        }
+
+        self.reserve(max_len);
+        let len = self.len();
+        let mut guard = Guard { vec: self, len };
+        unsafe { guard.vec.set_len(guard.len + max_len) }
```

Comment:
Rather than `unsafe` here can you use the same `resize` + `truncate` setup the `Vec<u8>` impl uses. My intent is to make this crate `forbid(unsafe_code)` other than cases where it is absolutely necessary like supporting `&mut str`.